### ISSUE TITLE
[trust problem] I can not trust simplex.chat when it send traffic to servers that is not deploy myself.

### ISSUE THREAD

**Issue by bronze1man** [simplex-chat/simplex-chat#1268]
I am a user in a team that care about privacy of logs of chat messages history and contracts .
I do not trust simplex.chat said in the first place, so I did a test:

I deploy two servers, ip a and ip b.
I config ip a to ios device c.
I config ip b to ios device d.
I put ios device a and device b into different nat , so it can not direct connect.
I find that device c can send message to device d.
So I am sure that are it send traffic to some hidden servers that is not deploy myself.
So I will not trust simplex.chat only base on those tests, I can not sure the servers that is not deploy myself can not read/log my message. without read and verify all source code myself.

Suggestion:
* Add a option that allow user connect to servers that deploy myself only. (I will use wireshark/tcpdump on router to verify it)

**Comment by bronze1man** [comment]
I know that ios push notifications stuff must use a server that is not deploy by myself. 
So I need options to disable it and document it (the domain or ip list) and document that code that do that stuff, so I can verify that code is not send my chat content to apple.

**Comment by epoberezkin** [comment]
> So I am sure that are it send traffic to some hidden servers that is not deploy by myself.

That is 100% not the case.
What seems likely to be happening is that the servers you configure only used for the new contacts you create, the contacts created before changing configuration continue using the servers they did.

You can check which servers are used to send and receive messages for each contact in the contact page (tap contact avatar on top of the chat). If it has one of simplex.im servers that would explain what you observe. If you have your servers there and you are certain that they are unreachable I'd love to investigate it further. I wasn't quite clear what is in different networks - for the test you describe it should be devices and servers, the devices would never communicate directly.

We will be adding a feature allowing to manually and automatically migrate to other servers - it is coming soon.

> Add a option that allow me connect to servers that deploy by myself only.

What you are asking is an option for "trusted servers" - we are considering it. We were thinking to only constrain the new connections though, not to prevent the existing contacts to work (as you seem to want), but it can be extended to be more restrictive and inform users that there are X contacts using non-trusted servers and that they would stop working.

**Comment by bronze1man** [comment]
> What seems likely to be happening is that the servers you configure only used for the new contacts you create, the contacts created before changing configuration continue using the servers they did.

No, in my tests, it looks like the configure of smp and webrtc address can not stop new contacts to use the default(hidden) servers . It may be a bug? (First configure to differents address, than add the contacts, then talks to eath others.)

**Comment by epoberezkin** [comment]
> @serrq: It seems correct to me that changes to servers also affect backwards those contacts who have established connections on public (or untrusted, I think you call them) servers.

You decide which message queue (think, like a mailbox) to use for each contact. If you want to move this mailbox to another server it requires some communication and agreement with that contact. Right now, it can only be done manually, by creating a new contact. v4.2 or maybe 4.3 will have experimental feature allowing you to switch contact to another server manually, by clicking a button. Later this year we will adding automatic migration of the contacts, and optionally it will be happening regularly too, but it cannot be instant - it will be happening over some short period of time.  And if the current server is broken then the automatic change won't be possible - you would have to send the new address out of band, in the same way the initial connection was made.

The expectation that changing the server should instantly change how you communicate with your contacts is based on the idea that each user has some global address and the servers are used to route messages between these addresses. But it's not how SimpleX is designed - users do not have any global addresses, and a server is part of the pair-wise address used to send or receive messages to one contact only, and changing this address requires some communication.

> @bronze1man: No, in my tests, it looks like the configure of smp and webrtc address can not stop new contacts to use the default(hidden) servers . It may be a bug? (First configure to differents address, than add the contacts, then talks to eath others.)

Have you changed the configuration on both clients before creating the connection? Your client only determines which servers will be used to receive messages, your contact's client determines which server is used by you to send messages (that is, both clients define the servers where they receive messages).

If after you change configuration on both clients the new connection is still established via the default servers, we need to look into it, but let's make a test first:

1. two clients with the empty profiles (so there are no old contacts that might still have traffic with the other servers)
2. change server configuration on both clients
3. make connection
4. make screenshot of contact pages - it would show which server is used to receive and which to send messages (see below)
5. also please check chat console in both clients for hostConnected events - these events would have servers when they are connected.

![image](https://user-images.githubusercontent.com/2769109/198898833-e6550336-3873-41e5-9691-20bde6dcba59.png)

**Comment by bronze1man** [comment]
> The expectation that changing the server should instantly change how you communicate with your contacts is based on the idea that each user has some global address and the servers are used to route messages between these addresses. 

No , my expectation is that I will setup my server after I download the app , and do not change it when the server is normal. I do not need that changing the server should instantly change how you communicate with your contacts is based on right now. I may need it when the server is broken , but not now.(I may expect two servers may solve this problem when one is broken)  When i change it , the old contacts are not important, i can always add my contact again face to face.

I will do the test again, later.

**Comment by epoberezkin** [comment]
There are ways to donate here: https://github.com/simplex-chat/simplex-chat#help-us-pay-for-3rd-party-security-audit

thank you!
